diff --git a/modules/cgal/test/test_memory_problem.cpp b/modules/cgal/test/test_memory_problem.cpp
index ece0b22115..7d61733a73 100644
--- a/modules/cgal/test/test_memory_problem.cpp
+++ b/modules/cgal/test/test_memory_problem.cpp
@@ -30,7 +30,7 @@ typedef CGAL::Fixed_alpha_shape_cell_base_3<Gt> Fb;
 typedef CGAL::Triangulation_data_structure_3<Vb, Fb> TDS;
 typedef CGAL::Regular_triangulation_3<Gt, TDS> Triangulation;
 #if CGAL_VERSION_NR > 1040911000
-typedef typename Gt::Point_3 Wpoint;
+typedef Gt::Point_3 Wpoint;
 #else
 typedef Gt::Point Wpoint;
 #endif
