diff --git a/modules/domino/src/assignment_containers.cpp b/modules/domino/src/assignment_containers.cpp
index 3344962202..6749e0f341 100644
--- a/modules/domino/src/assignment_containers.cpp
+++ b/modules/domino/src/assignment_containers.cpp
@@ -312,12 +312,12 @@ void ReadAssignmentContainer::add_assignment(const Assignment &) {
 
 ////////////////////////// RangeViewAssignmentContainer
 
-inline unsigned int RangeViewAssignmentContainer::get_number_of_assignments()
+unsigned int RangeViewAssignmentContainer::get_number_of_assignments()
     const {
   return end_ - begin_;
 }
 
-inline Assignment RangeViewAssignmentContainer::get_assignment(unsigned int i)
+Assignment RangeViewAssignmentContainer::get_assignment(unsigned int i)
     const {
   IMP_USAGE_CHECK(i < get_number_of_assignments(),
                   "Invalid assignment requested: " << i);
@@ -336,11 +336,11 @@ void RangeViewAssignmentContainer::add_assignment(const Assignment &) {
 
 ////////////////////////// HEAP ASSIGNMENT CONTAINER
 
-inline unsigned int HeapAssignmentContainer::get_number_of_assignments() const {
+unsigned int HeapAssignmentContainer::get_number_of_assignments() const {
   return d_.size();
 }
 
-inline Assignment HeapAssignmentContainer::get_assignment(unsigned int i)
+Assignment HeapAssignmentContainer::get_assignment(unsigned int i)
     const {
   IMP_USAGE_CHECK(i < get_number_of_assignments(),
                   "Invalid assignment requested: " << i);
