diff --git tools/build/tools/__init__.py tools/build/tools/__init__.py
index aed5834..ad101eb 100644
--- tools/build/tools/__init__.py
+++ tools/build/tools/__init__.py
@@ -546,6 +546,12 @@ def get_disabled_modules(extra_data_path, root="."):
             x, extra_data_path, root)["ok"]]
     )
 
+# Treat an open file as UTF8-encoded, regardless of the locale
+if sys.version_info[0] >= 3:
+    def open_utf8(fname, *args):
+        return open(fname, *args, encoding='UTF8')
+else:
+    open_utf8 = open
 
 _subprocesses = []
 
