diff --git modules/container/src/ConnectingPairContainer.cpp modules/container/src/ConnectingPairContainer.cpp
index cec3b50..be74b77 100644
--- modules/container/src/ConnectingPairContainer.cpp
+++ modules/container/src/ConnectingPairContainer.cpp
@@ -19,6 +19,13 @@
 #include <IMP/container_macros.h>
 #include <algorithm>
 
+// Work around Boost bug with adjacency_matrix in 1.60:
+// https://svn.boost.org/trac/boost/ticket/11880
+#include <boost/version.hpp>
+#if BOOST_VERSION == 106000
+# include <boost/type_traits/ice.hpp>
+#endif
+
 #include <boost/graph/kruskal_min_spanning_tree.hpp>
 #include <boost/graph/prim_minimum_spanning_tree.hpp>
 #include <boost/graph/adjacency_list.hpp>
diff --git modules/core/src/ConnectivityRestraint.cpp modules/core/src/ConnectivityRestraint.cpp
index 4cae519..5fa43e5 100644
--- modules/core/src/ConnectivityRestraint.cpp
+++ modules/core/src/ConnectivityRestraint.cpp
@@ -21,6 +21,13 @@
 
 #include <climits>
 
+// Work around Boost bug with adjacency_matrix in 1.60:
+// https://svn.boost.org/trac/boost/ticket/11880
+#include <boost/version.hpp>
+#if BOOST_VERSION == 106000
+# include <boost/type_traits/ice.hpp>
+#endif
+
 #include <boost/graph/kruskal_min_spanning_tree.hpp>
 #include <boost/graph/prim_minimum_spanning_tree.hpp>
 #include <boost/graph/adjacency_list.hpp>
diff --git modules/core/src/MSConnectivityRestraint.cpp modules/core/src/MSConnectivityRestraint.cpp
index 5b621c3..b5a8077 100644
--- modules/core/src/MSConnectivityRestraint.cpp
+++ modules/core/src/MSConnectivityRestraint.cpp
@@ -24,6 +24,13 @@
 
 #include <climits>
 
+// Work around Boost bug with adjacency_matrix in 1.60:
+// https://svn.boost.org/trac/boost/ticket/11880
+#include <boost/version.hpp>
+#if BOOST_VERSION == 106000
+# include <boost/type_traits/ice.hpp>
+#endif
+
 #include <boost/graph/kruskal_min_spanning_tree.hpp>
 #include <boost/graph/prim_minimum_spanning_tree.hpp>
 #include <boost/graph/adjacency_list.hpp>
diff --git modules/multifit/include/merge_tree_utils.h modules/multifit/include/merge_tree_utils.h
index 8154b4a..19b0036 100644
--- modules/multifit/include/merge_tree_utils.h
+++ modules/multifit/include/merge_tree_utils.h
@@ -8,6 +8,14 @@
 #define IMPMULTIFIT_MERGE_TREE_UTILS_H
 
 #include <boost/functional/hash.hpp>
+
+// Work around Boost bug with adjacency_matrix in 1.60:
+// https://svn.boost.org/trac/boost/ticket/11880
+#include <boost/version.hpp>
+#if BOOST_VERSION == 106000
+# include <boost/type_traits/ice.hpp>
+#endif
+
 #include <boost/graph/adjacency_matrix.hpp>
 #include <boost/graph/adjacency_list.hpp>
 #include <boost/pending/disjoint_sets.hpp>
diff --git modules/multifit/src/anchor_utilities.cpp modules/multifit/src/anchor_utilities.cpp
index a59110f..6d71701 100644
--- modules/multifit/src/anchor_utilities.cpp
+++ modules/multifit/src/anchor_utilities.cpp
@@ -10,6 +10,14 @@
 #include <IMP/algebra/vector_search.h>
 #include <IMP/atom/Atom.h>
 #include <IMP/atom/SecondaryStructureResidue.h>
+
+// Work around Boost bug with adjacency_matrix in 1.60:
+// https://svn.boost.org/trac/boost/ticket/11880
+#include <boost/version.hpp>
+#if BOOST_VERSION == 106000
+# include <boost/type_traits/ice.hpp>
+#endif
+
 #include <boost/graph/metric_tsp_approx.hpp>
 #include <boost/graph/adjacency_list.hpp>
 #include <boost/graph/adjacency_matrix.hpp>
diff --git modules/statistics/include/internal/TrivialPartitionalClustering.h modules/statistics/include/internal/TrivialPartitionalClustering.h
index 92aed35..72c42a1 100644
--- modules/statistics/include/internal/TrivialPartitionalClustering.h
+++ modules/statistics/include/internal/TrivialPartitionalClustering.h
@@ -11,8 +11,17 @@
 #include <IMP/statistics/statistics_config.h>
 #include "../PartitionalClustering.h"
 #include <IMP/check_macros.h>
+
+// Work around Boost bug with adjacency_matrix in 1.60:
+// https://svn.boost.org/trac/boost/ticket/11880
+#include <boost/version.hpp>
+#if BOOST_VERSION == 106000
+# include <boost/type_traits/ice.hpp>
+#endif
+
 #include <boost/graph/adjacency_matrix.hpp>
 #include <boost/graph/adjacency_list.hpp>
+
 IMPSTATISTICS_BEGIN_INTERNAL_NAMESPACE
 class TrivialPartitionalClustering : public PartitionalClustering {
   IMP::Vector<Ints> clusters_;
diff --git modules/statistics/include/internal/centrality_clustering.h modules/statistics/include/internal/centrality_clustering.h
index ddf8ab4..159047e 100644
--- modules/statistics/include/internal/centrality_clustering.h
+++ modules/statistics/include/internal/centrality_clustering.h
@@ -9,8 +9,17 @@
 #define IMPSTATISTICS_INTERNAL_CENTRALITY_CLUSTERING_H
 
 #include <IMP/statistics/statistics_config.h>
+
+// Work around Boost bug with adjacency_matrix in 1.60:
+// https://svn.boost.org/trac/boost/ticket/11880
+#include <boost/version.hpp>
+#if BOOST_VERSION == 106000
+# include <boost/type_traits/ice.hpp>
+#endif
+
 #include <boost/graph/adjacency_matrix.hpp>
 #include <boost/graph/adjacency_list.hpp>
+
 IMPSTATISTICS_BEGIN_NAMESPACE
 class PartitionalClustering;
 IMPSTATISTICS_END_NAMESPACE
diff --git modules/statistics/src/internal/centrality_clustering.cpp modules/statistics/src/internal/centrality_clustering.cpp
index dd147cc..62c8304 100644
--- modules/statistics/src/internal/centrality_clustering.cpp
+++ modules/statistics/src/internal/centrality_clustering.cpp
@@ -7,6 +7,14 @@
 #include <IMP/statistics/internal/centrality_clustering.h>
 #include <IMP/statistics/PartitionalClustering.h>
 #include <IMP/statistics/internal/TrivialPartitionalClustering.h>
+
+// Work around Boost bug with adjacency_matrix in 1.60:
+// https://svn.boost.org/trac/boost/ticket/11880
+#include <boost/version.hpp>
+#if BOOST_VERSION == 106000
+# include <boost/type_traits/ice.hpp>
+#endif
+
 #include <boost/property_map/vector_property_map.hpp>
 #include <boost/property_map/property_map.hpp>
 #include <boost/graph/adjacency_matrix.hpp>
