diff --git a/tools/build/make_python_init.py b/tools/build/make_python_init.py
index cf38fd5bb4f35dbdca333fb422d93c3da3ab7f01..a0bcec48eb9669e14ff8ea8d8032afd3fdfd475b 100755
--- a/tools/build/make_python_init.py
+++ b/tools/build/make_python_init.py
@@ -15,9 +15,11 @@ def build_init(module, source, infname, outfname):
             with open(infname) as inf:
                 shutil.copyfileobj(inf, outf)
 
-        # Add standard module functions
+        # Add standard module functions and constants
         outf.write("""
 
+__version__ = "%s"
+
 def get_module_version():
     '''Return the version of this module, as a string'''
     return "%s"
@@ -35,7 +37,7 @@ def get_example_path(fname):
     '''Return the full path to one of this module's example files'''
     import IMP
     return IMP._get_module_example_path("%s", fname)
-""" % (version, module.name, module.name, module.name))
+""" % (version, version, module.name, module.name, module.name))
 
 
 def main():
