diff --git a/modules/npctransport/Setup.cmake b/modules/npctransport/Setup.cmake
--- a/modules/npctransport/Setup.cmake
+++ b/modules/npctransport/Setup.cmake
@@ -7,13 +7,10 @@ add_custom_command(OUTPUT "${CMAKE_BINARY_DIR}/include/IMP/npctransport/internal
                           COMMAND protoc "--cpp_out=dllexport_decl=IMPNPCTRANSPORTEXPORT:."
                           "-I${CMAKE_SOURCE_DIR}/modules/npctransport/data/"
                           "${CMAKE_SOURCE_DIR}/modules/npctransport/data/npctransport.proto"
-                          COMMAND mv npctransport.pb.cc npctransport.pb.cpp
-                          # add config header to resolve export symbols
-                          COMMAND mv "${CMAKE_BINARY_DIR}/src/npctransport/npctransport.pb.h" "${CMAKE_BINARY_DIR}/src/npctransport/npctransport.pb.h.out"
-                          COMMAND echo "\"\#include\"" "\"<IMP/npctransport/npctransport_config.h>\"" > "${CMAKE_BINARY_DIR}/src/npctransport/npctransport.pb.h"
-                          COMMAND cat "${CMAKE_BINARY_DIR}/src/npctransport/npctransport.pb.h.out" >> "${CMAKE_BINARY_DIR}/src/npctransport/npctransport.pb.h"
-                          COMMAND rm -f "${CMAKE_BINARY_DIR}/include/IMP/npctransport/internal/npctransport.pb.h"
-                          COMMAND ln -s "${CMAKE_BINARY_DIR}/src/npctransport/npctransport.pb.h" "${CMAKE_BINARY_DIR}/include/IMP/npctransport/internal/npctransport.pb.h"
+                          COMMAND ${CMAKE_COMMAND} -E rename npctransport.pb.cc npctransport.pb.cpp
+                          COMMAND python "${CMAKE_SOURCE_DIR}/modules/npctransport/patch_protoc.py"
+                                         "${CMAKE_BINARY_DIR}/src/npctransport/npctransport.pb.h"
+                                         "${CMAKE_BINARY_DIR}/include/IMP/npctransport/internal/npctransport.pb.h"
                           DEPENDS "${CMAKE_SOURCE_DIR}/modules/npctransport/data/npctransport.proto"
                           COMMENT "Creating protoc stuff for npctransport"
                           WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/src/npctransport")
