diff --git a/modules/em2d/Setup.cmake b/modules/em2d/Setup.cmake
index e86962851e..822e84f6ac 100644
--- a/modules/em2d/Setup.cmake
+++ b/modules/em2d/Setup.cmake
@@ -1,4 +1,4 @@
-if("${OPENCV22_LIBRARIES}" STREQUAL "" AND "${OPENCV21_LIBRARIES}" STREQUAL "")
+if("${OPENCV3_LIBRARIES}" STREQUAL "" AND "${OPENCV22_LIBRARIES}" STREQUAL "" AND "${OPENCV21_LIBRARIES}" STREQUAL "")
   message(STATUS "Required dependency of OpenCV 2.1 or later not found")
 # disable em2d
   file(STRINGS ${CMAKE_BINARY_DIR}/data/build_info/disabled disabled)
diff --git a/modules/em2d/dependencies.py b/modules/em2d/dependencies.py
index 34d95a0908..f7cb2515f9 100644
--- a/modules/em2d/dependencies.py
+++ b/modules/em2d/dependencies.py
@@ -1,3 +1,3 @@
 required_modules = 'container:core:atom:algebra:em:display:gsl:domino'
 required_dependencies = 'FFTW3:Boost.ProgramOptions:Boost.FileSystem'
-optional_dependencies = 'OpenCV21:OpenCV22'
+optional_dependencies = 'OpenCV21:OpenCV22:OpenCV3'
diff --git a/modules/em2d/dependency/OpenCV22.description b/modules/em2d/dependency/OpenCV22.description
index a2d7991bb8..6265595f03 100644
--- a/modules/em2d/dependency/OpenCV22.description
+++ b/modules/em2d/dependency/OpenCV22.description
@@ -1,12 +1,12 @@
-full_name="OpenCV 2.2 or later"
+full_name="OpenCV 2.2 (or later versions of OpenCV 2)"
 pkg_config_name="opencv"
 libraries="opencv_core:opencv_imgproc:opencv_highgui:opencv_contrib"
 headers="opencv2/core/core.hpp:opencv2/imgproc/imgproc.hpp:opencv2/highgui/highgui.hpp:opencv2/core/version.hpp"
 body="""
-#if CV_MAJOR_VERSION>=2 && CV_MINOR_VERSION>=1
+#if CV_MAJOR_VERSION==2 && CV_MINOR_VERSION>=2
 new cv::Mat();
 #else
-#error "Version is not at least 2.1"
+#error "Version is not at least 2.2"
 #endif
 """
 versionheader="opencv2/core/version.hpp"
diff --git a/modules/em2d/dependency/OpenCV3.description b/modules/em2d/dependency/OpenCV3.description
new file mode 100644
index 0000000000..7df5816499
--- /dev/null
+++ b/modules/em2d/dependency/OpenCV3.description
@@ -0,0 +1,13 @@
+full_name="OpenCV 3 or later"
+pkg_config_name="opencv"
+libraries="opencv_core:opencv_imgproc:opencv_highgui:opencv_imgcodecs"
+headers="opencv2/core/core.hpp:opencv2/imgproc/imgproc.hpp:opencv2/highgui/highgui.hpp:opencv2/core/version.hpp"
+body="""
+#if CV_MAJOR_VERSION>=3
+new cv::Mat();
+#else
+#error "Version is not at least 3.0"
+#endif
+"""
+versionheader="opencv2/core/version.hpp"
+versioncpp="CV_MAJOR_VERSION:CV_MINOR_VERSION:CV_SUBMINOR_VERSION"
diff --git a/modules/em2d/include/opencv_interface.h b/modules/em2d/include/opencv_interface.h
index 9da595cf9b..e5f2740a4f 100644
--- a/modules/em2d/include/opencv_interface.h
+++ b/modules/em2d/include/opencv_interface.h
@@ -10,7 +10,7 @@
 #include "IMP/em2d/em2d_config.h"
 #include "IMP/algebra/Transformation2D.h"
 
-#if IMP_EM2D_HAS_OPENCV22
+#if IMP_EM2D_HAS_OPENCV22 || IMP_EM2D_HAS_OPENCV3
 #include "opencv2/core/core.hpp"
 #include "opencv2/core/version.hpp"
 #include "opencv2/imgproc/imgproc.hpp"
