diff --git a/modules/kernel/src/internal/directories.cpp b/modules/kernel/src/internal/directories.cpp
index bf1cb73963..b4f1117f70 100644
--- a/modules/kernel/src/internal/directories.cpp
+++ b/modules/kernel/src/internal/directories.cpp
@@ -38,8 +38,8 @@
 #endif
 
 IMPKERNEL_BEGIN_INTERNAL_NAMESPACE
-extern std::string imp_data_path;
-extern std::string imp_example_path;
+extern const char *imp_data_path;
+extern const char *imp_example_path;
 
 namespace {
 
diff --git a/tools/build/setup_paths.py b/tools/build/setup_paths.py
index b670b0b549..253d9fc07b 100755
--- a/tools/build/setup_paths.py
+++ b/tools/build/setup_paths.py
@@ -11,8 +11,8 @@ template = """
 #include <IMP/kernel_config.h>
 
 IMPKERNEL_BEGIN_INTERNAL_NAMESPACE
- std::string imp_data_path="%(datapath)s";
- std::string imp_example_path="%(examplepath)s";
+ const char *imp_data_path="%(datapath)s";
+ const char *imp_example_path="%(examplepath)s";
 IMPKERNEL_END_INTERNAL_NAMESPACE
 """
 
