diff --git a/tools/build/make_swig_wrapper.py b/tools/build/make_swig_wrapper.py
index b7a05a1475..c15801639a 100755
--- a/tools/build/make_swig_wrapper.py
+++ b/tools/build/make_swig_wrapper.py
@@ -108,6 +108,14 @@ def patch_file(infile, out, options):
         line = line.replace('"swig::SwigPyIterator *"', repl2)
         line = line.replace(orig, repl)
         line = line.replace("wrap.h-in", "wrap.h")
+        # work around internal compiler error with VS2008 building ISD
+        # and PMI wrappers
+        line = line.replace('new IMP::isd::MolecularDynamics()',
+                            'new IMP::isd::MolecularDynamics(nullptr)');
+        line = line.replace(
+             'new IMP::pmi::CompositeRestraint(arg1,arg2,arg3,arg4,arg5,arg6)',
+             'new IMP::pmi::CompositeRestraint(arg1,arg2,arg3,arg4,arg5,arg6,'
+             '"CompositeRestraint%1%")')
         # for some reason swig has issues with directors and VersionInfo
         # when %extend is used
         line = line.replace(" VersionInfo ", " IMP::VersionInfo ")
