Index: src/SConscript
===================================================================
--- src/SConscript	(revision 1064)
+++ src/SConscript	(working copy)
@@ -14,7 +14,7 @@
 
 # AIX and w32 link complain about undefined symbols unless we explicitly
 # link HDF5:
-if sys.platform == 'aix5':
+if sys.platform in ('aix5', 'darwin'):
     e.Append(LIBS=['hdf5', 'hdf5_hl'])
 if e['PLATFORM'] == 'win32':
     e.Append(SHLINKFLAGS=['hdf5.lib', 'hdf5_hl.lib'])
