diff --git a/modules/statistics/src/internal/centrality_clustering.cpp b/modules/statistics/src/internal/centrality_clustering.cpp
index 1c6242dfcf6513265015a98e88328cd17fbc5d1c..c5e5537d9c196689deb2978a4b86ad2b5b0200d0 100644
--- a/modules/statistics/src/internal/centrality_clustering.cpp
+++ b/modules/statistics/src/internal/centrality_clustering.cpp
@@ -32,11 +32,11 @@ using graph::has_no_edges;
 
 // Add workaround for https://github.com/boostorg/graph/issues/175
 // (Note that this may not be the right #if condition; we have only seen
-// the issue with Boost 1.71 through 1.74 on Macs with Homebrew, but other
+// the issue with Boost 1.71 through 1.75 on Macs with Homebrew, but other
 // systems may be affected.)
 #if defined(__clang__) && BOOST_VERSION >= 107100 && BOOST_VERSION <= 107200
 # include "our_bc_clustering.hpp"
-#elif defined(__clang__) && BOOST_VERSION >= 107300 && BOOST_VERSION <= 107400
+#elif defined(__clang__) && BOOST_VERSION >= 107300 && BOOST_VERSION <= 107500
 # include "our_bc_clustering_1_73.hpp"
 #else
 # include <boost/graph/bc_clustering.hpp>
