diff --git a/modules/atom/src/BrownianDynamics.cpp b/modules/atom/src/BrownianDynamics.cpp
index 93c8b3271e..8a03722b55 100644
--- a/modules/atom/src/BrownianDynamics.cpp
+++ b/modules/atom/src/BrownianDynamics.cpp
@@ -282,13 +282,21 @@ BrownianDynamics::do_step
   get_scoring_function()->evaluate(true);
   //  Ek = 0.0; // DEBUG: monitor kinetic energy
   //  M = 0.0; // DEBUG: monitor kinetic energy
+  /* Modern clang requires that we explicitly share constants, but older
+     clang reports an error if we try to do so */
   const unsigned int chunk_size = 5000;
   for (unsigned int b = 0; b < ps.size(); b += chunk_size) {
     IMP_TASK_SHARED(
-        (dt_fs, ikT, b), (ps),
+        (dt_fs, ikT, b),
+#if defined(__clang__) && __clang_major__ >= 10
+	(ps, chunk_size),
+#else
+	(ps),
+#endif
         do_advance_chunk(dt_fs, ikT, ps, b,
                          std::min<unsigned int>(b + chunk_size, ps.size()));
         , "brownian");
+
   }
   IMP_OMP_PRAGMA(taskwait)
   IMP_OMP_PRAGMA(flush)
