diff -Nur boost/config/compiler/visualc.hpp boost/config/compiler/visualc.hpp
--- boost/config/compiler/visualc.hpp	2014-10-26 05:36:42.000000000 -0700
+++ boost/config/compiler/visualc.hpp	2015-12-14 10:47:28.677130645 -0800
@@ -291,7 +291,7 @@
 
 //
 // last known and checked version is 19.00.22129 (VC14 CTP4):
-#if (_MSC_VER > 1800 && _MSC_FULL_VER > 190022129)
+#if (_MSC_VER > 1800 && _MSC_FULL_VER > 190023506)
 #  if defined(BOOST_ASSERT_CONFIG)
 #     error "Unknown compiler version - please run the configure tests and report the results"
 #  else
